### PR TITLE
PA1 Submission

### PR DESCRIPTION
**Pull request by rosesandhello**

### CODE REVIEW

File: fibmod.asm [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	      syscall               #print contents of save register four
+	      jal delimit
+	      
+	      add $s3, $s1, $s2     #add save register two and one, place result into save register four
+	      move $s1, $s2         #move contents of save register three into save register two
+	      move $s2, $s3         #move contents of save register four into save register three
+	      subi $s0, $s0, 1      #subtract one from the first save register	   
+	      
+	      bgtz $s0, loop	    #branch to loop if save register one is still greater than zero
+	      blez $0, exit
```
Comment (JShrein) [inline fibmod.asm:55]: Line 55 seems a bit odd to me.  Why test if $0 < 0 instead of using unconditional jump 'j'

File: fibmod.asm [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	      li $s3, 1	            #place one in save register four
+	      bgtz $s0, printOne    #branch to printOne if save register one has a value greater than zero
+	      
+printOne:     li $a0, 1		    #load one into the first argument register
+	      li $v0, 1		    #load one into the first system register
+	      syscall               #print one
+	      subi $s0, $s0, 1      #subtract one from the first save register
+	      jal delimit
+	      bgtz $s0, loop	    #branch to loop if save register one is still greater than zero
+
```
Comment (JShrein) [inline fibmod.asm:42]: -2 Should always test boundary cases.  If you enter a number < 2 the program still prints the first 2 fibonacci numbers.

File: tinycalc.asm [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	      add.d   $f6, $f2, $f4    # add doubles in f2 and f4, put result in f6
+	      sub.d   $f8, $f2, $f4    # subtract double in f4 from double in f2, put result in f8
+	      mul.d   $f10, $f2, $f4   # multiply the doubles in f2 and f4, put result in f10
+	      
+	      la   $a0, sum      # load address of intro into first argument register
+	      li   $v0, 4          # load print string call into first value register
+	      syscall		   # make system call that prints string prompt
+	      jal breakline
+	      
+	      mov.d $f12, $f6
```
Comment (JShrein) [inline tinycalc.asm:62]: -8 For this problem, you are required to store your calculation results in memory and retrieve them from memory for printing.

File: fifteenOdd.asm [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	.text
+	      li $s0, 1
+	      li $s2, 1
+	      li $s3, 15
+	    
+popOdd:
+	      addi $s2, $s2, 2
+	      add  $s0, $s0, $s2
+	      bne  $s2, $s3, popOdd
+
```
Comment (JShrein) [inline fifteenOdd.asm:22]: -6 Must store partial sum in memory

File: fifteenOdd.asm [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	      li $s0, 1
+	      li $s2, 1
+	      li $s3, 15
+	    
+popOdd:
+	      addi $s2, $s2, 2
+	      add  $s0, $s0, $s2
+	      bne  $s2, $s3, popOdd
+	      
+	      la $a0, ($s0)
```
Comment (JShrein) [inline fifteenOdd.asm:23]: -2 Line 23: This line works, but indirect addressing is an improper way to move a value to $a0.  By using ($s0) you're saying "load address of $s0", but $s0 is NOT an address, its a literal value.....It works but 'move' or 'add' would be more proper.

File: fifteenOdd.asm (hunk 4 above)
Comment (JShrein) [inline fifteenOdd.asm:22]: -2 Missing some comments

**Comment by JShrein** [comment]
SCORE: 80

PART 1:
Total: -2

Mostly Nice Work

Nice approach to this problem...very fast, very efficient.  

Line 55 seems a bit odd to me.  Why test if $0 < 0 instead of using unconditional jump 'j'

Would be even more awesome if no extra delimiter were printed

Should probably include an upper limit.  Even though there's no array to exceed the bounds of, the fibonacci numbers quickly grow beyond a signed 32 bit representation.  

-2 Should always test boundary cases.  If you enter a number < 2 the program still prints the first 2 fibonacci numbers.

PART 2:
Total: -8

Again, nice work.  You're using more advanced things than are required of you, but be careful to make sure you implement everything the instructions ask for.

-8 For this problem, you are required to store your calculation results in memory and retrieve them from memory for printing.

Youre conclusion message could probably use a "\n" but no points lost.

PART 3:
Total: -10

-6 Must store partial sum in memory

-2 Line 23: This line works, but indirect addressing is an improper way to move a value to $a0.  By using ($s0) you're saying "load address of $s0", but $s0 is NOT an address, its a literal value.....It works but 'move' or 'add' would be more proper.

-2 Missing some comments
